src/mrpro/operators/LinearOperator.py

```diff
@@ -47,6 +48,71 @@ def H(self) -> LinearOperator:  # noqa: N802
+
+        Parameters
+        ----------
+        initial_value
+            initial value to start the iteration; if chosen exactly as zero-tensor,
+            we randomly generate an initial value.
+        dim
+            the dimensions of the tensors on which the operator operates.
+            For example, for a matrix-vector multiplication example, a batched matrix tensor with shape (4,30,80,160),
+            input tensors of shape (4,30,160) to be multiplied, and dim = None, it is understood that the the
+            matrix representation of the operator corresponds to a block diagonal operator (with 4*30 matrices)
+            and thus the algorithm returns a tensor of shape (1,1,1) containing one single value.
+            In contrast, if for example, dim=(-1,), the algorithm computes a batched operator
+            norm and returns a tensor of shape (4,30,1) corresponding to the operator norms of the respective
+            matrices in the diagonal of the block-diagonal operator (if considered in matrix representation).
+            In any case, the output of the algorithm has the same number of dimensions as the elements of the
+            domain of the considered operator (whose dimensionality is implicitly defined by choosing dim), such that
+            the pointwise multiplication of the operator norm and elements of the domain (to be for example used
+            in a Landweber iteration) is well-defined.
+        max_iterations, optional
+            maximal number of iterations
+
+        Returns
+        -------
+            an estimaton of the operator norm
+        """
+        norm_initial_value = torch.linalg.vector_norm(initial_value, dim=dim, keepdim=True)
+        if dim is None:
+            if norm_initial_value == 0:
+                raise Exception('The initial value for the iteration should be different from the zero-vector.')
+        else:
+            if not (norm_initial_value > 0).all():
+                raise Exception(
+                    'Found at least one zero-vector as starting point. \
+                    For each of the considered operators, the initial value for the iteration \
+                    should be different from the zero-vector.'
+                )
+        if max_iterations < 1:
+            raise Exception('The number of iterations should be larger than zero.')
+
+        vector = initial_value
+        for _ in range(max_iterations):
+            # apply the operator to the vector
+            (vector,) = self.adjoint(*self(vector))
+
+            # normalize vector
+            vector /= torch.linalg.vector_norm(vector, dim=dim, keepdim=True)
+
+        (operator_vector,) = self.adjoint(*self(vector))
+
+        product = vector.real * operator_vector.real
+        if vector.is_complex() and operator_vector.is_complex():
+            product += vector.imag * operator_vector.imag
+        dim = tuple(dim) if dim is not None else dim
```

Comment:
Is tuple(dim) really needed here? does .sum() not accept any Sequence element?

Reply:
I had tried and got this error:

`src/mrpro/operators/LinearOperator.py:111: error: Argument 1 to "sum" of "TensorBase" has incompatible type "Sequence[int] | None"; expected "int | Size | list[int] | tuple[int, ...] | None"  [arg-type]`

Reply:
I will leave it as it is for the moment, but I have addressed the rest before my leave for the next few days.